torch/csrc/distributed/rpc/utils.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <torch/csrc/distributed/rpc/rpc_command_base.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+// Given an RPC message received as a request over the wire, deserialize it into
+// the appropriate 'RpcCommandBase' type.
+TORCH_API std::unique_ptr<RpcCommandBase> deserializeRequest(
```

Comment:
Yes, it helps in avoiding bugs where you might accidentally send a Response instead of Request or vice-versa. `deserializeRequest` only allows 'Request' types and `deserializeResponse` only allows 'Response' types.